homeassistant/components/scene/__init__.py

```diff
@@ -88,18 +89,35 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await component.async_unload_entry(entry)
 
 
-class Scene(Entity):
+class Scene(RestoreEntity):
     """A scene is a group of entities and the states we want them to be."""
 
-    @property
-    def should_poll(self) -> bool:
-        """No polling needed."""
-        return False
+    _attr_should_poll = False
+    __last_activated: datetime | None = None
 
     @property
+    @final
     def state(self) -> str | None:
         """Return the state of the scene."""
-        return STATE
+        if self.__last_activated is None:
+            return None
+        return self.__last_activated.isoformat()
```

Comment:
Should we store the isoformated timestamp in `__last_activated` since we don't use it for anything else?

Reply:
Done, will open a PR to do the same with the button